### ISSUE TITLE
Arduino 1.8.5 on macOS 10.13.1 and M2RET

### ISSUE THREAD

**Issue by martinseener** [collin80/M2RET#5]
Hi,
iam new to using the M2 to tinker around with my 2016 Skoda Octavia III and wanted to get M2RET running on my M2 with SavvyCAN at first.
I've installed the newest Arduino Software and also installed the Board Configuration described on the Macchina page and was also able to upload and test the "LED Blink" Example, so the basic things seem to work.

Then i "git clone"ed all the libraries here (due_can, wire, mcp...) to ~/Documents/Arduino/libraries with their normal "git clone" names. Then i tried to Compile the also latest master clone of M2RET and got the following errors (i think the double libraries found thing is not a problem). Can you help me?

Thanks, Martin

Error:

`Arduino: 1.8.5 (Mac OS X), Board: "Macchina M2"

/Users/martinseener/Documents/Arduino/M2RET/M2RET.ino: In function 'void setup()':
M2RET:310: error: 'class SWcan' has no member named 'setListenOnlyMode'
             SWCAN.setListenOnlyMode(true);
                   ^
M2RET:314: error: 'class SWcan' has no member named 'setListenOnlyMode'
             SWCAN.setListenOnlyMode(false);
                   ^
/Users/martinseener/Documents/Arduino/M2RET/M2RET.ino: In function 'void sendFrame(CAN_COMMON*, CAN_FRAME&)':
M2RET:392: error: comparison between distinct pointer types 'CAN_COMMON*' and 'SWcan*' lacks a cast
     if (bus == &SWCAN) whichBus = 2;
                 ^
/Users/martinseener/Documents/Arduino/M2RET/M2RET.ino: In function 'void loop()':
M2RET:705: error: no matching function for call to 'SWcan::GetRXFrame(CAN_FRAME&)'
     if (SWCAN.GetRXFrame(incoming)) {
                                  ^
/Users/martinseener/Documents/Arduino/M2RET/M2RET.ino:705:34: note: candidate is:
In file included from /Users/martinseener/Documents/Arduino/libraries/Single-Wire-CAN-mcp2515/src/MCP2515_sw_can.h:34:0,
                 from /Users/martinseener/Documents/Arduino/M2RET/M2RET.ino:37:
/Users/martinseener/Documents/Arduino/libraries/mcp2515/src/MCP2515.h:67:7: note: bool MCP2515::GetRXFrame(Frame&)
  bool GetRXFrame(Frame &frame);
       ^
/Users/martinseener/Documents/Arduino/libraries/mcp2515/src/MCP2515.h:67:7: note:   no known conversion for argument 1 from 'CAN_FRAME' to 'Frame&'
M2RET:959: error: cannot convert 'SWcan*' to 'CAN_COMMON*' for argument '1' to 'void sendFrame(CAN_COMMON*, CAN_FRAME&)'
                     if (out_bus == 2) sendFrame(&SWCAN, build_out_frame);
                                                                        ^
Multiple libraries were found for "due_can.h"
 Used: /Users/martinseener/Documents/Arduino/libraries/due_can
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/due_can
Multiple libraries were found for "Arduino_Due_SD_HSMCI.h"
 Used: /Users/martinseener/Documents/Arduino/libraries/M2_SD_HSMCI
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/M2_SD_HSMCI
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/M2_SD_HSMCI
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/M2_SD_HSMCI
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/M2_SD_HSMCI
Multiple libraries were found for "due_wire.h"
 Used: /Users/martinseener/Documents/Arduino/libraries/due_wire
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/due_wire
Multiple libraries were found for "lin_stack.h"
 Used: /Users/martinseener/Documents/Arduino/libraries/LIN
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/LIN
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/LIN
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/LIN
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/LIN
Multiple libraries were found for "MCP2515_sw_can.h"
 Used: /Users/martinseener/Documents/Arduino/libraries/Single-Wire-CAN-mcp2515
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/Single-Wire-CAN-mcp2515
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/Single-Wire-CAN-mcp2515
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/Single-Wire-CAN-mcp2515
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/Single-Wire-CAN-mcp2515
Multiple libraries were found for "MCP2515.h"
 Used: /Users/martinseener/Documents/Arduino/libraries/mcp2515
 Not used: /Users/martinseener/Library/Arduino15/packages/macchina/hardware/sam/0.2.1/libraries/mcp2515
exit status 1
'class SWcan' has no member named 'setListenOnlyMode'

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.
`

**Comment by collin80** [comment]
Did this ever get resolved? It seems you had multiple versions of the same library installed to different places.

**Comment by martinseener** [comment]
Nope didn't work. Stopped trying. What do you think i should do in order to fix it?

**Comment by collin80** [comment]
I know you didn't think that the double found libraries thing was a big deal but it is. You ideally want it to only find one copy of the library. But, even so, I thought that I had fixed this after you had the trouble. If you update all your library versions it just might work. The trouble was that I updated libraries without changing their version numbers so when it found two copies they'd have the same version # even if one was newer. This should be corrected. Now if you have two copies the newer one should get used automatically.
